src/rez/package_cache.py

```diff
@@ -166,12 +165,6 @@ def add_variant(self, variant, force=False):
                 % variant.uri
             )
 
-        if not os.path.isdir(variant_root):
```

Comment:
This seems to be the culprit for why the tests are breaking. We need to have a generic way to check if we can access the `variant_root` regardless of what type of PackageRepository is used (ex. database/filesystem).

Reply:
I've fixed the immediate issue by readding this check, but it's not fully fleshed out yet.